### PR TITLE
refactor: convert TS to new constraint setup 

### PR DESCRIPTION
**Pull request by jonaslagoni**
**Description**
This PR converts the TS generator to the new constraint setup introduced in https://github.com/asyncapi/modelina/pull/725

### CODE REVIEW

File: src/generators/typescript/constrainer/EnumConstrainer.ts [hunk 1]
```diff
@@ -27,40 +26,42 @@ export const DefaultEnumKeyConstraints: ModelEnumKeyConstraints = {
...
+  const constraints = {...DefaultEnumKeyConstraints, ...customConstraints};
+
+  return ({enumKey, enumModel, constrainedEnumModel}) => {
+    let constrainedEnumKey = enumKey;
+    constrainedEnumKey = constraints.NO_SPECIAL_CHAR(constrainedEnumKey);
+    constrainedEnumKey = constraints.NO_NUMBER_START_CHAR(constrainedEnumKey);
+    constrainedEnumKey = constraints.NO_EMPTY_VALUE(constrainedEnumKey);
+    constrainedEnumKey = constraints.NO_RESERVED_KEYWORDS(constrainedEnumKey);
+    constrainedEnumKey = constraints.NAMING_FORMATTER(constrainedEnumKey);
+    constrainedEnumKey = constraints.NO_DUPLICATE_KEYS(constrainedEnumModel, enumModel, constrainedEnumKey, constraints.NAMING_FORMATTER!);
```
Comment (Samridhi-98) [inline src/generators/typescript/constrainer/EnumConstrainer.ts:39]: Here getting a warning `forbidden not null assertion` .
Kindly disable it `/* eslint-disable  @typescript-eslint/no-non-null-assertion */` 🙂

File: src/generators/typescript/constrainer/PropertyKeyConstrainer.ts [hunk 2]
```diff
@@ -27,46 +26,18 @@ export const DefaultPropertyKeyConstraints: PropertyKeyConstraintOptions = {
...
-    constrainedPropertyKey = constraints.NO_EMPTY_VALUE!(constrainedPropertyKey);
-    constrainedPropertyKey = constraints.NO_RESERVED_KEYWORDS!(constrainedPropertyKey);
-    constrainedPropertyKey = constraints.NAMING_FORMATTER!(constrainedPropertyKey);
-    constrainedPropertyKey = constraints.NO_DUPLICATE_PROPERTIES!(constrainedObjectModel, objectModel, constrainedPropertyKey, constraints.NAMING_FORMATTER!);
+    constrainedPropertyKey = constraints.NO_SPECIAL_CHAR(constrainedPropertyKey);
+    constrainedPropertyKey = constraints.NO_NUMBER_START_CHAR(constrainedPropertyKey);
+    constrainedPropertyKey = constraints.NO_EMPTY_VALUE(constrainedPropertyKey);
+    constrainedPropertyKey = constraints.NO_RESERVED_KEYWORDS(constrainedPropertyKey);
+    constrainedPropertyKey = constraints.NAMING_FORMATTER(constrainedPropertyKey);
+    constrainedPropertyKey = constraints.NO_DUPLICATE_PROPERTIES(constrainedObjectModel, objectModel, constrainedPropertyKey, constraints.NAMING_FORMATTER!);
```
Comment (Samridhi-98) [inline src/generators/typescript/constrainer/PropertyKeyConstrainer.ts:40]: Here also getting a warning `forbidden not null assertion` .
Kindly disable it ` /* eslint-disable @typescript-eslint/no-non-null-assertion */ ` 🙂

**Review by Samridhi-98 — CHANGES_REQUESTED** [review]

File: src/generators/typescript/constrainer/EnumConstrainer.ts [hunk 3]
```diff
@@ -27,40 +26,42 @@ export const DefaultEnumKeyConstraints: ModelEnumKeyConstraints = {
   }
 };
 
-export type EnumKeyContext = {
-  enumKey: string,
-  constrainedEnumModel: ConstrainedEnumModel,
-  enumModel: EnumModel
+export function defaultEnumKeyConstraints(customConstraints?: Partial<ModelEnumKeyConstraints>): EnumKeyConstraint {
+  const constraints = {...DefaultEnumKeyConstraints, ...customConstraints};
```
Comment (magicmatatjahu) [inline src/generators/typescript/constrainer/EnumConstrainer.ts:30]:
```suggestion
export function defaultEnumKeyConstraints(customConstraints: Partial<ModelEnumKeyConstraints> = {}): EnumKeyConstraint {
  const constraints = {...DefaultEnumKeyConstraints, ...customConstraints};
```

File: src/generators/typescript/constrainer/ModelNameConstrainer.ts [hunk 4]
```diff
@@ -26,38 +25,16 @@ export const DefaultModelNameConstraints: ModelNameConstraints = {
...
-    }
-    if (customConstraints.NAMING_FORMATTER !== undefined) {
-      constraints.NAMING_FORMATTER = customConstraints.NAMING_FORMATTER;
-    }
-    if (customConstraints.NO_RESERVED_KEYWORDS !== undefined) {
-      constraints.NAMING_FORMATTER = customConstraints.NO_RESERVED_KEYWORDS;
-    }
-  }
+export function defaultModelNameConstraints(customConstraints?: Partial<ModelNameConstraints>): ModelNameConstraint {
+  const constraints = {...DefaultModelNameConstraints, ...customConstraints};
```
Comment (magicmatatjahu) [inline src/generators/typescript/constrainer/ModelNameConstrainer.ts:29]:
```suggestion
export function defaultModelNameConstraints(customConstraints: Partial<ModelNameConstraints> = {}): ModelNameConstraint {
  const constraints = {...DefaultModelNameConstraints, ...customConstraints};
```

File: src/generators/typescript/constrainer/PropertyKeyConstrainer.ts [hunk 5]
```diff
@@ -27,46 +26,18 @@ export const DefaultPropertyKeyConstraints: PropertyKeyConstraintOptions = {
     return NO_RESERVED_KEYWORDS(value, isReservedTypeScriptKeyword); 
   }
 };
+export function defaultPropertyKeyConstraints(customConstraints?: Partial<PropertyKeyConstraintOptions>): PropertyKeyConstraint {
+  const constraints = {...DefaultPropertyKeyConstraints, ...customConstraints};
```
Comment (magicmatatjahu) [inline src/generators/typescript/constrainer/PropertyKeyConstrainer.ts:30]:
```suggestion
export function defaultPropertyKeyConstraints(customConstraints: Partial<PropertyKeyConstraintOptions> = {}): PropertyKeyConstraint {
  const constraints = {...DefaultPropertyKeyConstraints, ...customConstraints};
```

**Review by magicmatatjahu — CHANGES_REQUESTED** [review]

File: src/generators/typescript/constrainer/EnumConstrainer.ts (hunk 1 above)
Comment (jonaslagoni) [reply src/generators/typescript/constrainer/EnumConstrainer.ts:39]: Good catch 👍 Guess that is what you get making changes 5am 😆

**Review by Samridhi-98 — APPROVED** [review]
LGTM 🚀

**Review by magicmatatjahu — APPROVED** [review]

**Review by magicmatatjahu — APPROVED** [review]

**Pull request merged**